database/src/postgres/tx_indexer.rs

```diff
@@ -0,0 +1,267 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use bigdecimal::num_traits::ToPrimitive;
+use bigdecimal::BigDecimal;
+use sqlx::QueryBuilder;
+
+#[async_trait]
+impl crate::base::tx_indexer::TxIndexerDbManager for crate::postgres::PostgresDBManager {
+    async fn create_tx_tables(&self) -> Result<()> {
+        // Create partitioned transactions table and partitions on each shard
+        for pool in self.shards_pool.values() {
+            // Create the partitioned transactions table
+            sqlx::query(
+                r#"
+                CREATE TABLE IF NOT EXISTS transactions (
+                    transaction_hash text NOT NULL,
+                    sender_account_id text NOT NULL,
+                    block_height numeric(20,0) NOT NULL,
+                    transaction_details bytea NOT NULL,
+                    PRIMARY KEY (transaction_hash)
+                ) PARTITION BY HASH (transaction_hash);
+                "#,
+            )
+            .execute(pool)
+            .await?;
+
+            // Create 100 partitions for the transactions table
+            sqlx::query(
+                r#"
+                DO $$
+                DECLARE
+                    i INT;
+                BEGIN
+                    FOR i IN 0..99 LOOP
+                        EXECUTE format(
+                            'CREATE TABLE IF NOT EXISTS transactions_%s PARTITION OF transactions FOR VALUES WITH (MODULUS 100, REMAINDER %s)', i, i
+                        );
+                    END LOOP;
+                END $$;
+                "#
+            )
+            .execute(pool)
+            .await?;
+        }
+
+        // Receipts, outcomes, and meta tables remain on meta_db_pool as before
+        sqlx::query(
+            "
+            CREATE TABLE IF NOT EXISTS receipts (
```

Comment:
Perhaps we should consider the possibility of splitting the receipts and outcomes tables on partitions by  receipt_id and outcome_id.

Reply:
I remind you that's this is a map firs of all, and whenever a user requests receipt they provide only the `receipt_id`, but no `sender_id` or other account information that can help us to identify the shard where the data is located.

Reply:
Yes, I remember about that, but I am talking about partitions, not shards.

Reply:
sorry, missed that, i agree